src/main/java/graphql/analysis/MaxQueryDepthInstrumentation.java

```diff
@@ -6,21 +6,46 @@
 import graphql.execution.instrumentation.SimpleInstrumentation;
 import graphql.execution.instrumentation.parameters.InstrumentationValidationParameters;
 import graphql.validation.ValidationError;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.List;
+import java.util.function.Function;
 
 import static graphql.execution.instrumentation.SimpleInstrumentationContext.whenCompleted;
 
 /**
- * Prevents execution if the query depth is greater than the specified maxDepth
+ * Prevents execution if the query depth is greater than the specified maxDepth.
+ *
+ * Use the {@link Function<QueryDepthInfo, Boolean>} parameter to supply a function to perform a custom action when the max depth is
+ * exceeded. If the function returns {@code true} a {@link AbortExecutionException} is thrown.
  */
 @PublicApi
 public class MaxQueryDepthInstrumentation extends SimpleInstrumentation {
 
+    private static final Logger log = LoggerFactory.getLogger(MaxQueryDepthInstrumentation.class);
+
     private final int maxDepth;
+    private final Function<QueryDepthInfo, Boolean> maxQueryDepthExceededFunction;
 
+    /**
+     * new Instrumentation
+     *
```

Comment:
That JavaDoc is strange.  I would rather words than leet.  

   Creates a new instrumentation  that tracks query depth

Reply:
I've updated the javadoc.